Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp

```diff
@@ -405,12 +405,32 @@ void JitArm64::fcmpX(UGeckoInstruction inst)
     ANDI2R(fpscr_reg, fpscr_reg, ~FPRF_MASK);
   }
 
+  ARM64Reg V0Q = ARM64Reg::INVALID_REG;
+  ARM64Reg V1Q = ARM64Reg::INVALID_REG;
+  if (upper_a)
+  {
+    V0Q = fpr.GetReg();
+    m_float_emit.DUP(singles ? 32 : 64, reg_encoder(V0Q), VA, 1);
+    VA = reg_encoder(V0Q);
+  }
+  if (upper_b)
```

Comment:
We have a `a == b` optimization below. I doubt that it will be called often, but we could easily skip the DUP call here.

Reply:
I don't see how we can skip the DUP call. We still need to perform the comparison, in case we have a NaN.

Reply:
Oh wait, you mean we can skip *one* DUP call. Yeah, that makes sense.

Reply:
Implemented.